### PR TITLE
The property of the parent element is used first when making a request

### PR DESCRIPTION
**Pull request by rocketweb**
### Description

Whenever you want to make an API request to an URL that has a variable in it, e.g. deleting a product requires the sku of that product in the URL, you would go the a 'meta' file and add an operation that looks something like this:

`<operation name="deleteProduct" dataType="product" type="delete" auth="adminOauth" url="/V1/products/{sku}" method="DELETE">
<contentType>application/json</contentType>
</operation>`

Adding `{sku}` to the url works if the entity that you want to delete is the only entity that has a property of that name, and there are no child entities associated with it that also have the property. In other words, if you want to create, for example, a bundle product, and be able to pass it the category and it's child products, like so:

> <createData entity="_defaultCategory" stepKey="createCategory"/>
            <createData entity="SimpleProductForBundle1" stepKey="createSimpleProduct1">
                <requiredEntity createDataKey="createCategory"/>
            </createData>
            <createData entity="SimpleProductForBundle2" stepKey="createSimpleProduct2">
                <requiredEntity createDataKey="createCategory"/>
            </createData>
            <createData entity="SimpleProductForBundle3" stepKey="createSimpleProduct3">
                <requiredEntity createDataKey="createCategory"/>
            </createData>

            <createData entity="BundleProduct" stepKey="createBundleProduct">
                <requiredEntity createDataKey="createCategory"/>
                <requiredEntity createDataKey="createSimpleProduct1"/>
                <requiredEntity createDataKey="createSimpleProduct2"/>
                <requiredEntity createDataKey="createSimpleProduct3"/>
            </createData>

When you want to delete the product by calling `<deleteData createDataKey="createBundleProduct" stepKey="deleteBundleProduct"/>`, which refferences something like `<operation name="deleteProductBundle" dataType="product_bundle" type="delete" auth="adminOauth" url="/V1/products/{sku}" method="DELETE">
        <contentType>application/json</contentType>
    </operation>` , it will actually delete the last entity that was passed as a parameter which has the 'sku' property. In the example above that would be "createSimpleProduct3". This is not how it should work.

The changes that I've made allows you to pick the entity that you want to make a refference to, when you want to call its property in the url. In order to delete a bundle product, one would add something like this in the metadata file:

```
<operation name="deleteProductBundle" dataType="product_bundle" type="delete" auth="adminOauth" url="/V1/products/{product_bundle.sku}" method="DELETE">
        <contentType>application/json</contentType>
    </operation>
```

Basically, instead of adding `{sku}`, one would add `{product_bundle.sku}`, where `product_bundle` is the metadata entity type, to the url.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

**Comment by KevinBKozan** [comment]
@rocketweb,
Following up with you about the license/CLA. If you could please sign the license/cla we can process this Pull Request.
